### ISSUE TITLE
Color transparency not working as expected

### ISSUE THREAD

**Issue by scottg521** [highcharts/highcharts#20381]
#### Expected behaviour
Bars will show up with opacity 0.1

#### Actual behaviour
Bars are initially full full color, and change to transparent only after mouseover
Also, YELLOW bar doesn't change (color name all caps) and 'lightgray' bar goes black (parser error)

I was modifying the demo from the Colors documentation to experiment with making colors transparent. This demo parses an array of color names to get the rgb colors, and adds a transparency as the 4th parameter from the color parser. The transparency doesn't seem to work correctly, though it does appear to dim the bars when you mouse over (and the full color doesn't come back). Also, there are some odd things when change to other color names--with capital letters, the initial color appears (YELLOW) but no dimming on hover; with 'lightgray' the initial color appears, but is not recognized on hover.

#### Live demo with steps to reproduce

https://jsfiddle.net/eLukrj9q/4/

Started from "Solid Colors" demo at https://www.highcharts.com/docs/chart-design-and-style/colors

#### Product version

current from code.highcharts.com

#### Affected browser(s)
Chrome (Mac), Edge (Windows)

**Comment by karolkolodziej** [comment]
Hi @scottg521!
Thank you for reporting the issue.

Colours are parsed properly when 
`YELLOW` -> `yellow` and 
`lightgray` -> `lightgrey`.
But this looks more like a problem with `1.4.0/rgbcolor.min.js` - [see available colours there.](https://cdn.jsdelivr.net/canvg/1.4.0/rgbcolor.min.js)
___

When it comes to the colours of the columns. Initially, the colour is applied explicitly from the option see:
https://github.com/highcharts/highcharts/blob/8ea315e85e335c9ed931d97521685ea09daf75ec/ts/Series/Column/ColumnSeries.ts#L718-L723
Later, when you hover over a point, the `pointAttribs` is triggered and colours are parsed there:
https://github.com/highcharts/highcharts/blob/8ea315e85e335c9ed931d97521685ea09daf75ec/ts/Series/Column/ColumnSeries.ts#L704-L710

@TorsteinHonsi 
Should we initially also parse the colours with `Color.parsers`?

**Comment by TorsteinHonsi** [comment]
@karolkolodziej @scottg521 

I think it's actually just a problem with the demo. The regex doesn't match upper case characters.

`regex: /^[a-z]+$/u,` should be `regex: /^[a-zA-Z]+$/u`, alternatively add an `i` flag.

If we fix this, it works as intended: https://jsfiddle.net/highcharts/0Lc2sudv/

**Comment by scottg521** [comment]
I agree that the issues with case and spelling are red herrings, though I'm not sure how you are supposed to know which color names are accepted ('lightgray' is on the [W3 list](https://www.w3.org/wiki/CSS/Properties/color/keywords), for example). 

The real issue, is that the opacity does not have an effect initially. It seems like this is a difference in parsing, as @karolkolodziej points out.

**Comment by karolkolodziej** [comment]
@TorsteinHonsi any thoughts on the initial color and its opacity?

**Comment by TorsteinHonsi** [comment]
@karolkolodziej Oh, I see. Didn't read the whole thread.

It looks like we can easily achieve that by modifying this line:

https://github.com/highcharts/highcharts/blob/ad751e9b3f49c04ee9ee1bc1ac27c3ea1e254821/ts/Series/Column/ColumnSeries.ts#L719

... to ...
`fill: color(fill).get(),`

It raises a couple of questions though.
* Will it affect performance? We could run some benchmarks with thousands, or tens of thousands of points. Since the color class is doing RegEx parsing it may be relevant. If it is slower, we could consider a pre-check for custom parsers.
* What about the stroke?

**Comment by scottg521** [comment]
From my perspective:
- it seems reasonable that if someone has many points, they should configure the most efficient way--i.e. using rgba(r,g,b,0.9) rather than 'red' or #CCC. Not sure if that avoids the need for a parser, but as long as there is an efficient option...
- seems like anything you set should take effect initially
